modules/daily_refetch/handle_periodic_operations.go

```diff
@@ -0,0 +1,72 @@
+package daily_refetch
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/forbole/juno/v3/parser"
+	"github.com/forbole/juno/v3/types/config"
+
+	"github.com/go-co-op/gocron"
+	"github.com/rs/zerolog/log"
+
+	parsecmdtypes "github.com/forbole/juno/v3/cmd/parse/types"
+)
+
+func (m *Module) RegisterPeriodicOperations(scheduler *gocron.Scheduler) error {
+	log.Debug().Str("module", "daily refetch").Msg("setting up periodic tasks")
+
+	// Setup a cron job to run every midnight
+	if _, err := scheduler.Every(1).Day().At("00:00").Do(func() {
+		m.refetchMissingBlocks()
+	}); err != nil {
+		return fmt.Errorf("error while setting up daily refetch periodic operation: %s", err)
+	}
+
+	return nil
+}
+
+// refetchMissingBlocks checks for missing blocks from one day ago and refetches them
+func (m *Module) refetchMissingBlocks() error {
+	log.Trace().Str("module", "daily refetch").Str("refetching", "blocks").
+		Msg("refetching missing blocks")
+
+	latestBlock, err := m.node.LatestHeight()
+	if err != nil {
+		return fmt.Errorf("error while getting latest block: %s", err)
+	}
+
+	blockCount, err := m.database.GetTotalBlocks()
+	if err != nil {
+		return fmt.Errorf("error while getting block count: %s", err)
+	}
+
+	blockHeightDayAgo, err := m.database.GetBlockHeightTimeDayAgo(time.Now())
+	if err != nil {
+		return fmt.Errorf("error while getting block height from a day ago: %s", err)
+	}
+	var startHeight int64 = blockHeightDayAgo.Height
+
+	if blockCount != latestBlock {
```

Comment:
Maybe we can instead check as below and put it to under line 42 for readability: 
```go
	if blockCount >= latestBlock {
		// Do nothing
		return nil
	}
```
(using `>=` because a new block may possibly be parsed and stored in DB during the time gap between line 34 and line 39.)